tests/ping.go

```diff
@@ -21,28 +21,59 @@ package tests
 
 import (
 	"fmt"
+	"regexp"
+	"strings"
 
 	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
 	"k8s.io/utils/net"
 
 	v1 "kubevirt.io/client-go/api/v1"
 )
 
+const promptExpression = "\\$ |\\# "
+const termNLExpression = "\r\n"
+
+var (
+	shellSuccess = regexp.MustCompile("\n0" + termNLExpression + ".*" + promptExpression)
+	shellFail    = regexp.MustCompile("\n[1-9].*" + termNLExpression + ".*" + promptExpression)
```

Comment:
Are you should failing `echo $?` always returns a numeric value?

Please take a look at -https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/3604/pull-kubevirt-e2e-kind-k8s-1.17.0-ipv6/1273898340039266304
That's what the expecter buffer contains after sending `echo $?` 
`echo $?\r\n-sh: can't create /etc/resolv.confnecho: Permission denied\r\n$ `

Reply:
That example was a `\n` escape issue, some command tried to write to the file `/etc/resolv.confnecho`.
`echo $?` will always print a numerical value (and a line feed).

Reply:
You're right, `echo $?` wasn't actually invoked in the above example.